### PR TITLE
ReadOnlyOption 支持请求级别

### PR DESCRIPTION
**Pull request by shihuili1218**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: jraft-core/src/main/resources/rpc.proto [hunk 1]
```diff
@@ -105,6 +105,7 @@ message ReadIndexRequest {
   required string server_id = 2;
   repeated bytes entries = 3;
   optional string peer_id = 4;
+  optional string readOnlyOptions = 5;
```
Comment (fengjiachun) [inline jraft-core/src/main/resources/rpc.proto:108]: 为什么不使用 enum ？

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java [hunk 2]
```diff
@@ -225,19 +228,36 @@ private void executeReadIndexEvents(final List<ReadIndexEvent> events) {
         if (events.isEmpty()) {
             return;
         }
-        final ReadIndexRequest.Builder rb = ReadIndexRequest.newBuilder() //
+        final ReadIndexRequest.Builder readLogRequestBuilder = ReadIndexRequest.newBuilder() //
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java:231]: 变量名好像不需要改？

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java [hunk 3]
```diff
@@ -225,19 +229,36 @@ private void executeReadIndexEvents(final List<ReadIndexEvent> events) {
...
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java:232]: 下面这段能不能再抽个方法，看起来收集 readlease 和 readsafe 是重复的逻辑

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java [hunk 4]
```diff
@@ -318,35 +339,41 @@ public void join() throws InterruptedException {
...
-                      throw new OverloadException(errorMsg);
+                    if (!this.readIndexQueue.tryPublishEvent(translator)) {
+                        final String errorMsg = "Node is busy, has too many read-index requests, queue is full and bufferSize=" + this.readIndexQueue.getBufferSize();
+                        ThreadPoolsFactory.runClosureInThread(this.node.getGroupId(), closure,
+                                new Status(RaftError.EBUSY, errorMsg));
+                        this.nodeMetrics.recordTimes("read-index-overload-times", 1);
+                        LOG.warn("Node {} ReadOnlyServiceImpl readIndexQueue is overload.", this.node.getNodeId());
+                        if (closure == null) {
+                            throw new OverloadException(errorMsg);
+                        }
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java:374]: 确认一下，这里应该只做了代码格式化对吗

Comment (shihuili1218) [reply jraft-core/src/main/java/com/alipay/sofa/jraft/core/ReadOnlyServiceImpl.java:374]: 是的。仅做了format

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**